### PR TITLE
- fix remaining order iterations

### PR DESCRIPTION
**Pull request by wenq1**

### CODE REVIEW

File: quanttrader/strategy/strategy_base.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def place_order(self, o):
         """
         o.source = self.id         # identify source
         if o.create_time is None:
-            o.create_time = datetime.now().strftime('%H:%M:%S.%f')
+            o.create_time = datetime.now().strftime('%Y-%m-%d %H:%M:%S.%f')
```
Comment (letianzj) [inline quanttrader/strategy/strategy_base.py:97]: I actually use place_order for daily real trade; So to make the trade blotter window cleaner, I didn't add the date. I use adjust_position for backtest trades, that's why it has dates in the multi-day backtest setting. But I guess it does not hurt to be consistent that both have dates in the timestamp.

File: quanttrader/strategy/strategy_base.py [hunk 2]
```diff
@@ -113,13 +113,11 @@ def adjust_position(self, sym, size_from, size_to, timestamp=None):
...
         o.order_size = size_to - size_from
-        o.source = self.id  # identify source
-        if timestamp is None:
-            o.create_time = datetime.now().strftime('%Y-%m-%d %H:%M:%S.%f')
-        else:
+        if timestamp is not None:
             o.create_time = timestamp
-        if (self.active):
-            self.strategy_manager.place_order(o)
+
```
Comment (letianzj) [inline quanttrader/strategy/strategy_base.py:118]: On the trade window there is a strategy control window to activate/pause/stop a strategy. This is managed by StrategyManager. So if this strategy is inactive, then StrategyManager will not place the order.

File: quanttrader/strategy/strategy_base.py [hunk 3]
```diff
@@ -113,13 +113,11 @@ def adjust_position(self, sym, size_from, size_to, timestamp=None):
         o.full_symbol = sym
         o.order_type = OrderType.MARKET
         o.order_size = size_to - size_from
-        o.source = self.id  # identify source
-        if timestamp is None:
-            o.create_time = datetime.now().strftime('%Y-%m-%d %H:%M:%S.%f')
-        else:
+        if timestamp is not None:
```
Comment (letianzj) [inline quanttrader/strategy/strategy_base.py:116]: Shouldn't we stamp an order if it does not have one?

**Review by letianzj — COMMENTED** [review]
Thanks for the pull request. The conditional order matching in backtest is valid. Timestamp arguably should be consistent. If you only use adjust_position for backtest, like me, it is ok to not check the active status and submit directly without StrategyManager.

File: quanttrader/strategy/strategy_base.py (hunk 3 above)
Comment (wenq1) [reply quanttrader/strategy/strategy_base.py:116]: It's is already done in the callee, i.e. place_order (). Hence no need to do that in adjust_position ()

File: quanttrader/strategy/strategy_base.py (hunk 2 above)
Comment (wenq1) [reply quanttrader/strategy/strategy_base.py:118]: This is also done place_order (). adjust_position () will just call place_order ()

File: quanttrader/strategy/strategy_base.py (hunk 1 above)
Comment (wenq1) [reply quanttrader/strategy/strategy_base.py:97]: Just to make adjust_position and place_order consistent. adjust_position can just call place_order. 
place_order will check for active order and place it in strategy manager conditionally.

**Comment by wenq1** [comment]
The key point of this PR is not about redundancy removal (e3e66633ddc2cf6bdb6898da734afdab668ad7d3). It is about the dictionary comprehension issue (439799bcc438c8f41cb6744e551a45fbb85331d5)

I don't know the python version you are using, but check out
https://stackoverflow.com/questions/3420122/filter-dict-to-contain-only-certain-keys


In backtest_brokerage.py:

```
# Either of the following lines will work:
self._active_orders = {k: v for k, v in self._active_orders.items () if k in _remaining_active_orders_id}
self._active_orders = {k: self._active_orders[k] if k in _remaining_active_orders_id}

# The following line will NOT work:
self._active_orders = {k: v for k, v in self._active_orders if k in _remaining_active_orders_id}
```

File: quanttrader/strategy/strategy_base.py (hunk 3 above)
Comment (letianzj) [reply quanttrader/strategy/strategy_base.py:116]: I was adding `adjust_postion` as a shortcut to place market order. It does have some redundency checks.

**Review by letianzj — APPROVED** [review]
fix remaining order iterations in matching engine. remove some redundancy codes in `adjust_position`.

**Pull request merged**